Gordon360/Services/RecIM/MatchService.cs

```diff
@@ -0,0 +1,411 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Models.ViewModels.RecIM;
+using Gordon360.Models.CCT.Context;
+using Microsoft.Extensions.Configuration;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Match = Gordon360.Models.CCT.Match;
+using Microsoft.Graph;
+
+namespace Gordon360.Services.RecIM
+{
+    public class MatchService : IMatchService
+    {
+        private readonly CCTContext _context;
+        private readonly IAccountService _accountService;
+
+
+        public MatchService(CCTContext context,  IAccountService accountService)
+        {
+            _context = context;
+            _accountService = accountService;
+        }
+
+        public MatchViewModel GetSimpleMatchViewByID(int matchID)
+        {
+            var res = _context.Match.FirstOrDefault(m => m.ID == matchID);
+            return res;
+        }
+
+        public IEnumerable<LookupViewModel> GetMatchLookup(string type)
+        {
+            switch (type)
+            {
+                case "status":
+                {
+                    var res = _context.MatchStatus
+                        .Select(s => new LookupViewModel
+                        {
+                            ID = s.ID,
+                            Description = s.Description
+                        })
+                        .AsEnumerable();
+                    return res;
+                }
+                case "teamstatus":
+                {
+                    var res = _context.MatchTeamStatus
+                        .Select(s => new LookupViewModel
+                        {
+                            ID = s.ID,
+                            Description = s.Description
+                        })
+                        .AsEnumerable();
+                    return res;
+                }
+                case "surface":
+                {
+                    var res = _context.Surface
+                        .Select(s => new LookupViewModel
+                        {
+                            ID = s.ID,
+                            Description = s.Description
+                        })
+                        .AsEnumerable();
+                    return res;
```

Comment:
This is a lot of almost identical code and identical data structure. Have you built these lookup tables individually? It would be much cleaner to have a single 'code table' that serves all of these lookups.

Reply:
A single table has a lot of drawbacks, mostly with relationships. If we have one single lookup table, than it is technically valid for, e.g. `Match.StatusID` to point to a row in the lookup table that is intended for `ParticipantStatus`.

They be the same schema, but that doesn't mean they are the same data. It's much better practice, IMO, to keep them as separate tables.

You could, theoretically, unify all these methods using reflection, but I'm not sure it's worth it. I would even argue it's better to keep all of these as separate types instead of having one single `LookupViewModel`.

Reply:
We should probably take this conversation offline until we agree so we don't pollute this comment thread with our argument :).

Reply:
Alright, I'll leave this unresolved until there is something agreed upon